### DISCUSSION TITLE
Call to FT_ERROR_STRING causes procedure not found error during runtime

### DISCUSSION THREAD

**Discussion by ShayTheSquiddog** [ImageMagick/ImageMagick#7647] [Development] [answered]
### ImageMagick version

7.1.1-38

### Operating system

Windows

### Operating system, version and so on

Windows 11, mingw64

### Description

The call to "FT_ERROR_STRING" in [annotate.c](https://github.com/search?q=repo%3AImageMagick/ImageMagick%20FT_Error_String&type=code) causes an error in magick++ during the initial launch of any c++ program using a magick++ object or function. This problem is only remedied by rebuilding ImageMagick with `--with-freetype=no`.

This occurs even when Freetype is rebuilt and installed with the line `#define FT_CONFIG_OPTION_ERROR_STRINGS` re-enabled (it is disabled by default). The version of freetype I'm using is 2.13.3.

### Steps to Reproduce

Build ImageMagick with freetype enabled. In any c++ code, include the Magick++ library and add any line of code that will create a  new object from Magick++ or uses one of it's functions. i.e. `Magick::Image image;` or `InitializeMagick();`. build and attempt to run the executable. The program should build without issue however when running the executable, it will throw an error saying:
```The procedure entry point FT_ERROR_STRING could not be located in the dynamic link library C:\msys64\mingw64\bin\libMagickCore-7.Q16HDRI-10.dll```

This error will not appear if executed from the terminal, and it will instead self-terminate without throwing any error.

### Images

![Screenshot 2024-09-23 164307](https://github.com/user-attachments/assets/f0396c91-87ec-49e0-9531-5af3d302acbf)

**Comment by dlemstra** [comment]
This feels like an issue in your build setup and not something that is causing by our library. We are using a C++ compiler for the binaries of the windows build and with have no issue with both the static and dynamic build.

**Reply by ShayTheSquiddog** [reply]
I'm not entirely sure what would be the issue in this case then. I've uninstalled and reinstalled the entirety of mingw and performed the following set of commands:
```UCRT64 ~ pacman -S --needed base-devel mingw-w64-ucrt-x86_64-toolchain
MINGW64 ~ pacman -S mingw-w64-x86_64-curl
MINGW64 ~ pacman -S mingw-w64-x86_64-imagemagick
MINGW64 ~ pacman -S mingw-w64-x86_64-ghostscript mingw-w64-x86_64-graphviz mingw-w64-x86_64-libjxl mingw-w64-x86_64-librsvg mingw-w64-x86_64-openexr
MINGW64 ~ pacman -R mingw-w64-x86_64-imagemagick
MINGW64 ~ pacman -S mingw-w64-x86_64-imagemagick
```

Even with this full reinstall, it continues to give me the same issue.
Strangely enough, it throws a similar error but with libraqm as well.
![image](https://github.com/user-attachments/assets/5bd788c2-7eb0-4527-9f81-7e0e4ff56b01)

**Reply by ShayTheSquiddog** [reply, marked as answer]
Nevermind, I figured it out. After comparing the setup between my desktop (which works fine) and my laptop (which wasn't working), I found out the problem. I needed to add `C:\msys64\mingw64\bin` to my system path. I didn't think it to be something that I needed to do, but I guess it was (even though `C:\msys64\ucrt64\bin` is in my PATH)

**Reply by dlemstra** [reply]
Thanks for coming back here and sharing how your resolved this!

**Reply by ShayTheSquiddog** [reply]
No worries! I've seen plenty of threads recording problems on different forums end up with just a "I fixed it" followed by the thread being closed. Figured I should probably share in case someone else ends up with a similar problem.
